src/Microsoft.Framework.PackageManager/SourceControl/SourceCommand.cs

```diff
@@ -0,0 +1,244 @@
+                return false;
+            }
+
+            _solutionRoot = ProjectResolver.ResolveRootDirectory(projectFile);
+
+            var packagesFolder = ResolvePackagesFolder();
+            if (string.IsNullOrEmpty(packagesFolder))
+            {
+                return false;
+            }
+
+            var packageVersion = ResolvePackageVersion(projectFile);
+            if (packageVersion == null)
+            {
+                return false;
+            }
+
+            var packageResolver = new DefaultPackagePathResolver(packagesFolder);
+            var packageFolder = packageResolver.GetPackageDirectory(_packageId, packageVersion);
+            packageFolder = Path.Combine(packagesFolder, packageFolder);
+
+            var snapshotInfo = ReadRepositoryInfo(packageFolder);
+            if (snapshotInfo == null)
+            {
+                return false;
+            }
+
+            string repoType;
+            if (!snapshotInfo.TryGetValue(Constants.RepoTypeKey, out repoType))
+            {
+                _reports.WriteError("Repository type information is missing from the repository information file.");
+                return false;
+            }
+
+            var provider = SourceControlProviderFactory.ResolveProvider(repoType, _reports);
+            if (provider == null)
+            {
+                _reports.WriteError($"Unknown repository type '{repoType}'");
+                return false;
+            }
+
+            var sourcesFolder = ResolveSourcesFolder();
+            var sourceFolderName = provider.CreateShortFolderName(snapshotInfo);
+            var sourceDestinationFullPath = Path.Combine(sourcesFolder, sourceFolderName);
+
+            if (!Directory.Exists(sourceDestinationFullPath))
+            {
+                _reports.WriteInformation($"Downloading sources in '{sourceDestinationFullPath}'...");
+                if (!provider.GetSources(sourceDestinationFullPath, snapshotInfo))
+                {
+                    return false;
+                }
+            }
+            else
+            {
+                _reports.WriteInformation($"Sources already found in '{sourceDestinationFullPath}'");
+            }
+
+            var srcFolder = provider.GetSourceFolderPath(snapshotInfo);
+            srcFolder = Path.Combine(sourceDestinationFullPath, srcFolder);
+            if (!Directory.Exists(srcFolder))
+            {
+                _reports.WriteError($"The source code folder '{srcFolder}' is missing.");
+                return false;
+            }
+
+            var globalFile = GlobalSettings.GetGlobalFilePath(_solutionRoot);
+            if (!File.Exists(globalFile))
+            {
+                _reports.WriteError($"The '{GlobalSettings.GlobalFileName}' is missing from '{_solutionRoot}'.");
+                return false;
+            }
+
+            _reports.Verbose.WriteLine($"Updating {GlobalSettings.GlobalFileName}...");
+            ModifyJson(globalFile, jObj =>
```

Comment:
This seems like an odd pattern to me. Why not put the lambda body in the `ModifyJson` method (and give it a more appropriate name, like `AddSourcesDirectoryToProjects` or something...)? The `ModifyJson` method is only ever used here and given this lambda body.